numpy/testing/_private/utils.py

```diff
@@ -386,7 +386,7 @@ def assert_equal(actual, desired, err_msg='', verbose=True):
             if not signbit(desired) == signbit(actual):
                 raise AssertionError(msg)
 
-    except (TypeError, ValueError, NotImplementedError):
+    except (TypeError, ValueError, NotImplementedError, DeprecationWarning):
```

Comment:
This seems risky, and at least deserving of a source comment.

Reply:
Added an explanation and a version changed

Reply:
It's tempting to add an entirely separate `except DeprecationWarning` with this comment

Reply:
Thinking some more, we should not be catching warnings as exceptions at all - shouldn't we be suppressing them even if they're not exceptions (or not suppressing them at all)?